include/libdnf/base/goal.hpp

```diff
@@ -54,53 +54,285 @@ class Goal {
     explicit Goal(const libdnf::BaseWeakPtr & base);
     explicit Goal(libdnf::Base & base);
     ~Goal();
+
+    // TODO(jmracek) Implemented not yet
     void add_module_enable(const std::string & spec);
-    /// Prevent reinstallation by adding of already installed packages with the same NEVRA
+
+    /// Add install job to the goal for future resolvement. The operation will not result in package reinstall when a
+    /// requested package is already installed. In default it uses `clean_requirements_on_remove` with `false` value.
```

Comment:
> By default uses `clean_requirements_on_remove` set to `false`.

Though this feels like you've just picked a single (semi-)arbitrary value from the settings and decided to describe it here, what about the other ones?

Reply:
I picked `clean_requirements_on_remove` for a good reason. When setting == GoalSetting::AUTO it is expected that the value is taken from `ConfigMain`. In case of `clean_requirements_on_remove` it behaves differently for particular methods.